### PR TITLE
Fix fullscreen not working + menu refactor (fix #1206)

### LINKED ISSUE #1206: Cannot make Nativefier apps full-screen on macOS
**Homework**

- [x] I looked at `nativefier --help` and https://github.com/nativefier/nativefier/blob/master/API.md
- [x] I checked the [Troubleshooting section of the README](https://github.com/nativefier/nativefier/blob/master/README.md#troubleshooting).
- [x] I searched existing issues, open & closed. Yes, my bug is new.
- [x] I'm using the latest version available at https://github.com/nativefier/nativefier/releases

**Bug description**

The "View" > "Toggle Full Screen" does not work on Nativefier applications.  Nor does using Exposé and dragging the window to the full-screen window bar.

**Steps to reproduce**

Give your ***full*** nativefier command and its logs, with the ***`--verbose` flag***, on a ***public*** site:

<details><summary>Nativefier Build Log</summary>

```
$ build-nativefier-apps ebeip90
Discord@ebeip90.app -> https://discord.com/app
######################################################################################################################################################################################################### 100.0%
Running in verbose mode! This will produce a mountain of logs and is recommended only for troubleshooting or if you like Shakespeare.

Performing async options post-processing.
Got icon from options. Using it, no inferring needed
Got name Discord@ebeip90.app from options. No inferring needed
Sanitized filename for Discord@ebeip90.app : Discord@ebeip90app
options {
  packager: {
    appCopyright: undefined,
    appVersion: undefined,
    arch: 'x64',
    asar: false,
    buildVersion: undefined,
    darwinDarkModeSupport: true,
    dir: '/usr/local/Cellar/nativefier/43.1.3/libexec/lib/node_modules/nativefier/app',
    electronVersion: '12.0.7',
    icon: 'icon.png',
    name: 'Discord@ebeip90app',
    out: '/private/var/folders/45/y3yh_pws51qd8dfx007hf6br0000gn/T/tmp.8ta699zx',
    overwrite: false,
    platform: 'darwin',
    portable: false,
    targetUrl: 'https://discord.com/app',
    tmpdir: false,
    upgrade: false,
    upgradeFrom: undefined,
    win32metadata: {
      ProductName: 'Discord@ebeip90.app',
      InternalName: 'Discord@ebeip90.app',
      FileDescription: 'Discord@ebeip90.app'
    }
  },
  nativefier: {
    accessibilityPrompt: true,
    alwaysOnTop: false,
    backgroundColor: null,
    basicAuthPassword: null,
    basicAuthUsername: null,
    blockExternalUrls: false,
    bookmarksMenu: null,
    bounce: true,
    browserwindowOptions: undefined,
    clearCache: false,
    counter: false,
    crashReporter: undefined,
    disableContextMenu: undefined,
    disableDevTools: undefined,
    disableGpu: false,
    diskCacheSize: null,
    disableOldBuildWarning: false,
    enableEs3Apis: false,
    fastQuit: false,
    fileDownloadOptions: undefined,
    flashPluginDir: null,
    fullScreen: false,
    globalShortcuts: null,
    hideWindowFrame: undefined,
    ignoreCertificate: false,
    ignoreGpuBlacklist: false,
    inject: [],
    insecure: false,
    internalUrls: '(.*?\\.(google|facebook|messenger)\\.com.*)',
    lang: undefined,
    maximize: false,
    nativefierVersion: '43.1.3',
    processEnvs: undefined,
    proxyRules: null,
    showMenuBar: false,
    singleInstance: true,
    titleBarStyle: null,
    tray: false,
    userAgent: 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/90.0.4430.93 Safari/537.36 Edg/90.0.818.56',
    userAgentOverriden: true,
    verbose: true,
    versionString: undefined,
    width: 1280,
    height: 800,
    minWidth: undefined,
    minHeight: undefined,
    maxWidth: undefined,
    maxHeight: undefined,
    widevine: false,
    x: undefined,
    y: undefined,
    zoom: 1
  }
}

Preparing Electron app...
Copying electron app from /usr/local/Cellar/nativefier/43.1.3/libexec/lib/node_modules/nativefier/app to /var/folders/45/y3yh_pws51qd8dfx007hf6br0000gn/T/nativefier-15-58-39-app--91953-wBrno21h7Dof
Writing app config to /var/folders/45/y3yh_pws51qd8dfx007hf6br0000gn/T/nativefier-15-58-39-app--91953-wBrno21h7Dof/nativefier.json
No files to inject, skipping copy.
Updating /var/folders/45/y3yh_pws51qd8dfx007hf6br0000gn/T/nativefier-15-58-39-app--91953-wBrno21h7Dof/package.json 'name' field to discord@ebeip90app-nativefier-18566a

Converting icons...
Converting icon icon.png to /var/folders/45/y3yh_pws51qd8dfx007hf6br0000gn/T/nativefier-15-58-39-iconconv--91953-BTyR1BlUlaLQ/icon.icns. Calling shell command: "/usr/local/Cellar/nativefier/43.1.3/libexec/lib/node_modules/nativefier/icon-scripts/convertToIcns" "icon.png" "/var/folders/45/y3yh_pws51qd8dfx007hf6br0000gn/T/nativefier-15-58-39-iconconv--91953-BTyR1BlUlaLQ/icon.icns"
Conversion succeeded and produced icon at /var/folders/45/y3yh_pws51qd8dfx007hf6br0000gn/T/nativefier-15-58-39-iconconv--91953-BTyR1BlUlaLQ/icon.icns
Copying icons if necessary
No copying necessary on macOS; aborting

Packaging... This will take a few seconds, maybe minutes if the requested Electron isn't cached yet...
  electron-packager Electron Packager 15.2.0
  electron-packager Node v16.2.0
  electron-packager Host Operating system: darwin 20.6.0 (x64) +0ms
  electron-packager Packager Options: {"arch":"x64","asar":false,"darwinDarkModeSupport":true,"dir":"/var/folders/45/y3yh_pws51qd8dfx007hf6br0000gn/T/nativefier-15-58-39-app--91953-wBrno21h7Dof","electronVersion":"12.0.7","icon":"/var/folders/45/y3yh_pws51qd8dfx007hf6br0000gn/T/nativefier-15-58-39-iconconv--91953-BTyR1BlUlaLQ/icon.icns","name":"Discord@ebeip90app","out":"/private/var/folders/45/y3yh_pws51qd8dfx007hf6br0000gn/T/tmp.8ta699zx","overwrite":false,"platform":"darwin","portable":false,"targetUrl":"https://discord.com/app","tmpdir":false,"upgrade":false,"win32metadata":{"ProductName":"Discord@ebeip90.app","InternalName":"Discord@ebeip90.app","FileDescription":"Discord@ebeip90.app"}} +1ms
  electron-packager Target Platforms: darwin +0ms
  electron-packager Target Architectures: x64 +0ms
  electron-packager Inferring appVersion from version in /var/folders/45/y3yh_pws51qd8dfx007hf6br0000gn/T/nativefier-15-58-39-app--91953-wBrno21h7Dof/package.json +0ms
  electron-packager Application name: Discord@ebeip90app +12ms
  electron-packager Target Electron version: 12.0.7 +0ms
  electron-packager Ignored path regular expressions: [
  '/package-lock\\.json$',
  '/yarn\\.lock$',
  '/\\.git($|/)',
  '/node_modules/\\.bin($|/)',
  '\\.o(bj)?$'
] +0ms
  electron-packager Downloading Electron with options {"platform":"darwin","arch":"x64","version":"12.0.7","artifactName":"electron"} +0ms
Packaging app for platform darwin x64 using electron v12.0.7
  electron-packager Creating /private/var/folders/45/y3yh_pws51qd8dfx007hf6br0000gn/T/tmp.8ta699zx/darwin-x64-template +14ms
  electron-packager Extracting /Users/zachriggle/Library/Caches/electron/4f029753a9a8576806dced687079cdf26638acb9ef0f41141be4eff390c1bd52/electron-v12.0.7-darwin-x64.zip to /private/var/folders/45/y3yh_pws51qd8dfx007hf6br0000gn/T/tmp.8ta699zx/darwin-x64-template +0ms
  electron-packager Initializing app in /private/var/folders/45/y3yh_pws51qd8dfx007hf6br0000gn/T/tmp.8ta699zx/Discord@ebeip90app-darwin-x64 from /private/var/folders/45/y3yh_pws51qd8dfx007hf6br0000gn/T/tmp.8ta699zx/darwin-x64-template template +0ms
  electron-packager Ignored paths based on the out param: [
  '/private/var/folders/45/y3yh_pws51qd8dfx007hf6br0000gn/T/tmp.8ta699zx/Discord@ebeip90app-darwin-x64',
  '/private/var/folders/45/y3yh_pws51qd8dfx007hf6br0000gn/T/tmp.8ta699zx/Discord@ebeip90app-darwin-arm64',
  '/private/var/folders/45/y3yh_pws51qd8dfx007hf6br0000gn/T/tmp.8ta699zx/Discord@ebeip90app-linux-ia32',
  '/private/var/folders/45/y3yh_pws51qd8dfx007hf6br0000gn/T/tmp.8ta699zx/Discord@ebeip90app-linux-x64',
  '/private/var/folders/45/y3yh_pws51qd8dfx007hf6br0000gn/T/tmp.8ta699zx/Discord@ebeip90app-linux-armv7l',
  '/private/var/folders/45/y3yh_pws51qd8dfx007hf6br0000gn/T/tmp.8ta699zx/Discord@ebeip90app-linux-arm64',
  '/private/var/folders/45/y3yh_pws51qd8dfx007hf6br0000gn/T/tmp.8ta699zx/Discord@ebeip90app-linux-mips64el',
  '/private/var/folders/45/y3yh_pws51qd8dfx007hf6br0000gn/T/tmp.8ta699zx/Discord@ebeip90app-mas-x64',
  '/private/var/folders/45/y3yh_pws51qd8dfx007hf6br0000gn/T/tmp.8ta699zx/Discord@ebeip90app-mas-arm64',
  '/private/var/folders/45/y3yh_pws51qd8dfx007hf6br0000gn/T/tmp.8ta699zx/Discord@ebeip90app-win32-ia32',
  '/private/var/folders/45/y3yh_pws51qd8dfx007hf6br0000gn/T/tmp.8ta699zx/Discord@ebeip90app-win32-x64',
  '/private/var/folders/45/y3yh_pws51qd8dfx007hf6br0000gn/T/tmp.8ta699zx/Discord@ebeip90app-win32-arm64'
] +2s
  electron-packager Copying icon "/var/folders/45/y3yh_pws51qd8dfx007hf6br0000gn/T/nativefier-15-58-39-iconconv--91953-BTyR1BlUlaLQ/icon.icns" to app's Resources as "electron.icns" +0ms
  electron-packager Renaming Electron to Discord@ebeip90app in /private/var/folders/45/y3yh_pws51qd8dfx007hf6br0000gn/T/tmp.8ta699zx/Discord@ebeip90app-darwin-x64/Electron.app/Contents/MacOS +38ms
  electron-packager Renaming Electron Helper to Discord@ebeip90app Helper in /private/var/folders/45/y3yh_pws51qd8dfx007hf6br0000gn/T/tmp.8ta699zx/Discord@ebeip90app-darwin-x64/Electron.app/Contents/Frameworks/Electron Helper.app/Contents/MacOS +1ms
  electron-packager Renaming Electron Helper (Renderer) to Discord@ebeip90app Helper (Renderer) in /private/var/folders/45/y3yh_pws51qd8dfx007hf6br0000gn/T/tmp.8ta699zx/Discord@ebeip90app-darwin-x64/Electron.app/Contents/Frameworks/Electron Helper (Renderer).app/Contents/MacOS +0ms
  electron-packager Renaming Electron Helper (Plugin) to Discord@ebeip90app Helper (Plugin) in /private/var/folders/45/y3yh_pws51qd8dfx007hf6br0000gn/T/tmp.8ta699zx/Discord@ebeip90app-darwin-x64/Electron.app/Contents/Frameworks/Electron Helper (Plugin).app/Contents/MacOS +0ms
  electron-packager Renaming Electron Helper (GPU) to Discord@ebeip90app Helper (GPU) in /private/var/folders/45/y3yh_pws51qd8dfx007hf6br0000gn/T/tmp.8ta699zx/Discord@ebeip90app-darwin-x64/Electron.app/Contents/Frameworks/Electron Helper (GPU).app/Contents/MacOS +1ms
  electron-packager Renaming Electron Helper.app to Discord@ebeip90app Helper.app in /private/var/folders/45/y3yh_pws51qd8dfx007hf6br0000gn/T/tmp.8ta699zx/Discord@ebeip90app-darwin-x64/Electron.app/Contents/Frameworks +0ms
  electron-packager Renaming Electron Helper (Renderer).app to Discord@ebeip90app Helper (Renderer).app in /private/var/folders/45/y3yh_pws51qd8dfx007hf6br0000gn/T/tmp.8ta699zx/Discord@ebeip90app-darwin-x64/Electron.app/Contents/Frameworks +0ms
  electron-packager Renaming Electron Helper (Plugin).app to Discord@ebeip90app Helper (Plugin).app in /private/var/folders/45/y3yh_pws51qd8dfx007hf6br0000gn/T/tmp.8ta699zx/Discord@ebeip90app-darwin-x64/Electron.app/Contents/Frameworks +0ms
  electron-packager Renaming Electron Helper (GPU).app to Discord@ebeip90app Helper (GPU).app in /private/var/folders/45/y3yh_pws51qd8dfx007hf6br0000gn/T/tmp.8ta699zx/Discord@ebeip90app-darwin-x64/Electron.app/Contents/Frameworks +0ms

Finalizing build...
App built to /private/var/folders/45/y3yh_pws51qd8dfx007hf6br0000gn/T/tmp.8ta699zx/Discord@ebeip90app-darwin-x64, move to wherever it makes sense for you and run the app bundle.
```
</details>

**Expected behavior**

Should be able to make apps full-screen.

**Actual behavior**

Cannot make apps full-screen.

**Context**

```
$ nativefier --version
43.1.3

$ node --version
v16.2.0

$ npm --version
7.13.0

$ sw_vers
ProductName:	macOS
ProductVersion:	11.5
BuildVersion:	20G5023d
```

### PR DESCRIPTION
**Pull request by TheCleric**
Fixes #1206

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -101,14 +101,13 @@
       "<rootDir>/src.*",
       "<rootDir>/node_modules.*",
       "<rootDir>/app/src.*",
-      "<rootDir>/app/dist.*",
```
Comment (ronjouch) [inline package.json:104]: Good catch. I had a valuable intention, though: to exclude the webpack output folder (which is just packing of tsc's output, useless to test/watch, so let's not test/watch). So, pls keep the line but replace `dist` with `lib`.

File: package.json [hunk 2]
```diff
@@ -101,14 +101,13 @@
...
       "<rootDir>/app/node_modules.*"
     ],
     "watchPathIgnorePatterns": [
       "<rootDir>/src.*",
       "<rootDir>/tsconfig.json",
+      "<rootDir>/app/node_modules.*",
       "<rootDir>/app/src.*",
-      "<rootDir>/app/dist.*",
```
Comment (ronjouch) [inline package.json:111]: Same here, pls replace `dist` with `lib`

File: package.json [hunk 3]
```diff
@@ -101,14 +101,13 @@
       "<rootDir>/src.*",
       "<rootDir>/node_modules.*",
       "<rootDir>/app/src.*",
-      "<rootDir>/app/dist.*",
       "<rootDir>/app/node_modules.*"
     ],
     "watchPathIgnorePatterns": [
       "<rootDir>/src.*",
       "<rootDir>/tsconfig.json",
+      "<rootDir>/app/node_modules.*",
```
Comment (ronjouch) [inline package.json:109]: Did you encounter "too many watch handles" errors on your machine?

- If no, let's not ignore if it's not necessary for the watcher to function. Might be nice when battling a dependency bug to have a working live reload on deps as you fiddle with the dependency bug
- If yes, please exclude `"<rootDir>/node_modules.*",` too

File: app/src/helpers/windowHelpers.ts [hunk 4]
```diff
@@ -137,16 +137,16 @@ export function getDefaultWindowOptions(
     ...(options.browserwindowOptions?.webPreferences ?? {}),
   };
 
-  const defaultOptions = {
-    // Convert dashes to spaces because on linux the app name is joined with dashes
-    title: options.name,
+  const defaultOptions: BrowserWindowConstructorOptions = {
+    fullscreenable: true,
```
Comment (ronjouch) [inline app/src/helpers/windowHelpers.ts:141]: 👍

File: app/src/components/mainWindow.ts [hunk 5]
```diff
@@ -180,30 +167,6 @@ function setupCounter(
...
-    goToURL,
-    openExternal,
-    zoomBuildTimeValue: options.zoom,
-    zoomIn,
-    zoomOut,
-    zoomReset,
-  };
-
-  createMenu(menuOptions);
-}
```
Comment (ronjouch) [inline app/src/components/mainWindow.ts:205]: 👍 thanks for the de-weirdification 😄

File: app/src/components/menu.ts [hunk 6]
```diff
@@ -100,11 +132,7 @@ export function createMenu({
     submenu: [
       {
         label: 'Back',
-        accelerator: (() => {
-          const backKbShortcut =
-            process.platform === 'darwin' ? 'Cmd+Left' : 'Alt+Left';
-          return backKbShortcut;
-        })(),
+        accelerator: isOSX() ? 'CmdOrAlt +Left' : 'Alt+Left',
```
Comment (ronjouch) [inline app/src/components/menu.ts:135]:
```suggestion
        accelerator: isOSX() ? 'CmdOrAlt+Left' : 'Alt+Left',
```

File: app/src/components/menu.ts [hunk 7]
```diff
@@ -132,27 +156,32 @@ export function createMenu({
       },
       {
         label: 'Reload',
-        accelerator: 'CmdOrCtrl+R',
```
Comment (ronjouch) [inline app/src/components/menu.ts:135]: Did you lose the accelerator on this one?

File: app/src/components/menu.ts [hunk 8]
```diff
@@ -132,27 +156,32 @@ export function createMenu({
...
         type: 'separator',
       },
       {
         label: 'Toggle Full Screen',
-        accelerator: (() => {
-          if (process.platform === 'darwin') {
-            return 'Ctrl+Cmd+F';
+        accelerator: isOSX() ? 'Ctrl+Cmd+F' : 'F11',
+        enabled: mainWindow.isFullScreenable() || isOSX(),
+        visible: mainWindow.isFullScreenable() || isOSX(),
```
Comment (ronjouch) [inline app/src/components/menu.ts:168]: Neato

File: app/src/components/menu.ts [hunk 9]
```diff
@@ -287,7 +321,7 @@ export function createMenu({
         {
           label: 'Quit',
           accelerator: 'Cmd+Q',
-          click: appQuit,
+          role: 'quit',
```
Comment (ronjouch) [inline app/src/components/menu.ts:324]: For my info: what's the nuance between `click: appQuit` and `role: 'quit'` ?

File: app/src/components/menu.ts [hunk 10]
```diff
@@ -305,55 +339,58 @@ export function createMenu({
...
 
+  return menuTemplate;
+}
+
+function injectBookmarks(menuTemplate: MenuItemConstructorOptions[]): void {
+  const bookmarkConfigPath = path.join(__dirname, '..', 'bookmarks.json');
+
+  if (!fs.existsSync(bookmarkConfigPath)) {
+    return;
+  }
```
Comment (ronjouch) [inline app/src/components/menu.ts:350]: :heart: early exits

File: app/src/components/menu.ts [hunk 11]
```diff
@@ -305,55 +339,58 @@ export function createMenu({
...
+      fs.readFileSync(bookmarkConfigPath, 'utf-8'),
+    );
+    const bookmarksMenu: MenuItemConstructorOptions = {
+      label: bookmarksMenuConfig.menuLabel,
+      submenu: bookmarksMenuConfig.bookmarks.map((bookmark) => {
+        switch (bookmark.type) {
+          case 'link':
             if (!('title' in bookmark && 'url' in bookmark)) {
-              throw Error(
+              throw new Error(
```
Comment (ronjouch) [inline app/src/components/menu.ts:362]: Surprise ESLint didn't chime about this. Can we enable a rule to make it error?

**Review by ronjouch — APPROVED** [review]
LGTM except a few nits & questions. Letting you address or ignore as you please and feel free to merge

File: app/src/components/menu.ts (hunk 7 above)
Comment (TheCleric) [reply app/src/components/menu.ts:135]: Nope. That's the nice thing about `role`, it provides he accelerator and the function for you for common menu functions:

Screen Shot 2021-06-04 at 10 37 58 AM

File: app/src/components/menu.ts (hunk 9 above)
Comment (TheCleric) [reply app/src/components/menu.ts:324]: See above on Reload, but `role` passes the need for accelerators and functions to Electron to handle.

File: app/src/components/menu.ts (hunk 11 above)
Comment (TheCleric) [reply app/src/components/menu.ts:362]: Unfortunately I don't see a rule that covers this:

https://eslint.org/docs/rules/

File: app/src/components/menu.ts (hunk 9 above)
Comment (ronjouch) [reply app/src/components/menu.ts:324]: Neat, didn't know. Thx 👍.

**Review by ronjouch — APPROVED** [review]

**Pull request merged**